src/imitation/scripts/train_adversarial.py

```diff
@@ -257,8 +278,17 @@ def plot_disc_loss(self):
     plt.legend()
     self._savefig("plot_fight_loss_disc", self.show_plots)
 
-  def add_data_ep_reward(self, env, name):
+  def add_data_ep_reward(self, epoch):
     """Calculate and record average episode returns."""
+    if epoch in self.ep_reward_X:
+      # Don't calculate ep reward twice.
+      return
+    self.ep_reward_X.append(epoch)
+    self._add_data_ep_reward(self.venv_norm_obs, "Ground Truth Reward")
```

Comment:
We do rollouts once for each reward function? This feels inefficient: the policy is the same in all cases. It's also higher variance, since the rollouts will be different in each case.

Can we do a rollout just once in the ground truth, and then compute the returns under each of the models?

Reply:
Good point. I've rewritten this section to use the same rollouts repeatedly.